spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/config/LoadBalancerAutoConfiguration.java

```diff
@@ -50,10 +54,14 @@
 @ConditionalOnProperty(value = "spring.cloud.loadbalancer.enabled", havingValue = "true", matchIfMissing = true)
 public class LoadBalancerAutoConfiguration {
 
+	private static final String ZONE_SPLITTER_COMMA = ",";
+
 	@Bean
 	@ConditionalOnMissingBean
 	public LoadBalancerZoneConfig zoneConfig(Environment environment) {
-		return new LoadBalancerZoneConfig(environment.getProperty("spring.cloud.loadbalancer.zone"));
+		return new LoadBalancerZoneConfig(environment.getProperty("spring.cloud.loadbalancer.zone"),
+				List.of(Optional.ofNullable(environment.getProperty("spring.cloud.loadbalancer.secondary-zones"))
+						.orElse(StringUtil.EMPTY_STRING).split(ZONE_SPLITTER_COMMA)));
```

Comment:
Spring's `StringUtils` has some tokenize methods that are better to use here.

Reply:
Changes split method into `StringUtils.commaDelimitedListToStringArray`